src/XCCDF/result.c

```diff
@@ -1067,14 +1080,40 @@ xmlNode *xccdf_target_identifier_to_dom(const struct xccdf_target_identifier *ti
 	}
 }
 
-xmlNode *xccdf_rule_result_to_dom(struct xccdf_rule_result *result, xmlDoc *doc, xmlNode *parent, const struct xccdf_version_info* version_info)
+xmlNode *xccdf_rule_result_to_dom(struct xccdf_rule_result *result, xmlDoc *doc, xmlNode *parent, const struct xccdf_version_info* version_info, struct xccdf_benchmark *benchmark, bool use_stig_rule_id)
 {
+	const char *idref = xccdf_rule_result_get_idref(result);
+	if (use_stig_rule_id) {
+		// Don't output rules with no stig ids
+		if (!idref || !benchmark)
+			return;
+
+		struct xccdf_item *item = xccdf_benchmark_get_member(benchmark, XCCDF_RULE, idref);
+		if (!item)
+			return;
+
+		const char *stig_rule_id = NULL;		
+		struct oscap_reference_iterator *references = xccdf_item_get_references(XRULE(item));
+		while (oscap_reference_iterator_has_more(references)) {
+			struct oscap_reference *ref = oscap_reference_iterator_next(references);
+			if (strcmp(ref->href, DISA_STIG_VIEWER_HREF) == 0) {
+				stig_rule_id = ref->title;
+				break;
+			}
+		}
+		oscap_reference_iterator_free(references);
+
+		if (!stig_rule_id)
```

Comment:
Shouldn't criteria for not outputting the rule result be no `stig_rule_id` and rule result `notchecked`? See my comments in previous review.

Reply:
Leys say we a have a checked rule with no stig id, what is the point in keeping it on the Stig Viewer output? We would just get the popup complaining about it.

Reply:
Regarding the documentation, there is a little notice in manual.adoc

Reply:
Motivation for this is to make it easier to spot mistakes in DISA Profiles.

> Leys say we a have a checked rule with no stig id, what is the point in keeping it on the Stig Viewer output?

In this scenario, somebody might have forgotten to assign a STIG ID to this Rule, or it may have been selected by mistake.